packages/react/src/components/radio-button/radio-button.tsx

```diff
@@ -0,0 +1,127 @@
+    display: inline-block;
+    height: var(--size-1x);
+    margin: 0;
+    position: relative;
+    width: var(--size-1x);
+    
+    ${(theme) => focus(theme, { selector: '+' })}
+     
+    &:checked {
+        border: 2px solid ${({ theme, disabled }) => (disabled ? theme.component['radio-button-disabled-border-color'] : theme.component['radio-button-checked-border-color'])};
+        &::after {
+            background-color: ${({ theme, disabled }) => (disabled ? theme.component['radio-button-disabled-border-color'] : theme.component['radio-button-checked-background-color'])};
+            border-radius: 50%;
+            content: '';
+            display: block;
+            height: var(--size-half);
+            left: 50%;
+            position: absolute;
+            top: 50%;
+            transform: translate(-50%, -50%);
+            width: var(--size-half);
+        }
+    }
+        
+    &:hover {
+        border: 1px solid ${({ theme, disabled }) => (disabled ? theme.component['radio-button-disabled-hover-border-color'] : theme.component['radio-button-hover-border-color'])};
+    }
+`;
+
+interface StyledLabelProps {
+    disabled?: boolean;
+    hasLabel: boolean;
+}
+
+const StyledLabel = styled.label<StyledLabelProps>`
+    font-size: 0.875rem;
+    line-height: 1.5rem;
+    margin-left: var(--spacing-1x);
+`;
+
+const StyledContainer = styled.div`
+    align-items: center;
+    display: inline-flex;
+    margin-top: var(--spacing-1x);
+    position: relative;
+`;
+
+interface RadioButtonProps {
+    ariaLabel?: string;
+    ariaLabelledBy?: string[];
+    checked?: boolean;
+    className?: string
+    defaultChecked?: boolean;
+    disabled?: boolean;
+    externalLabelId?: string;
+    id?: string;
+    label?: string;
+    name?: string;
+    value?: string;
+    visuallyHidden?: boolean;
+
+    onChange?(event: ChangeEvent<HTMLInputElement>): void;
+}
+
+export const RadioButton: FunctionComponent<RadioButtonProps> = ({
+    ariaLabel,
+    ariaLabelledBy,
+    checked,
+    className,
+    defaultChecked,
+    disabled,
+    externalLabelId,
+    id,
+    label,
+    name,
+    value,
+    visuallyHidden,
+    onChange,
+}) => {
+    const externalLabel = externalLabelId ? document.getElementById(externalLabelId)?.innerText : '';
+    const inputLabel = externalLabelId ? `${label} ${externalLabel}` ?? '' : label;
+
+    return (
+        <StyledContainer>
+            <StyledInput
+                data-testid={`radiobutton-${id}`}
+                id={id}
+                type="radio"
+                name={name}
+                className={className}
+                value={value}
+                defaultChecked={defaultChecked}
+                checked={checked}
+                disabled={disabled}
+                onChange={onChange}
+            />
+            <StyledLabel
```

Comment:
Le `<label>` devrait uniquement s'afficher s'il est explicitement défini par l'utilisateur.
De plus, il serait préférable que le `<label>` reprenne par défaut l'`id` de l'élément `<input>`. Ca éviterait aux développeurs de devoir l'ajouter manuellement et réduirait ainsi le risque d'oubli, ce qui pourrait compromettre l'accessibilité de l'interface.

À valider avec les autres devs mais tu pourrais avoir:
```suggestion
import { useId } from '../../hooks/use-id';
 ...
const generatedId = useId();
const inputId = id || generatedId;
...
<StyledInput
  data-testid={`radiobutton-${inputId}`}
  id={inputId}
... 
 {label && (
	<StyledLabel
		data-testid={`radiobutton-${inputId}_label`}
		htmlFor={inputId}
		className={`${className}_label`}
		disabled={disabled}
	>
	{
		visuallyHidden
		? <VisuallyHidden>{inputLabel}</VisuallyHidden>
		: inputLabel
	}
	</StyledLabel>
)}
```